R/test_wrapper.R

```diff
@@ -542,34 +547,40 @@ exp_normality<- function(df, ..., n_sample = 50) {
     df.qqline <- data.frame()
     df.model <- data.frame()
 　  for (col in selected_cols) {
-　    # set plot.it to FALSE to disable plotting (avoid launching another window)
-　    res <- stats::qqnorm(df[[col]], plot.it=FALSE)
-　    df.qq <- dplyr::bind_rows(df.qq, data.frame(x=res$x, y=res$y, col=col))
-
-      # bind reference line data too.
-      ref_res <- qqline_data(df[[col]])
-      min_x <- min(res$x, na.rm=TRUE)
-      max_x <- max(res$x, na.rm=TRUE)
-      ref_min_y <- ref_res[[1]] + ref_res[[2]] * min_x
-      ref_max_y <- ref_res[[1]] + ref_res[[2]] * max_x
-　    df.qqline <- dplyr::bind_rows(df.qqline, data.frame(x=c(min_x, max_x), refline_y=c(ref_min_y,ref_max_y), col=col))
-
-      if (n_sample > 5000) {
-        n_sample <- 5000 # shapiro.test takes only up to max of 5000 samples. 
-      }
-
-      if (length(df[[col]]) > n_sample) {
-        col_to_test <- sample(df[[col]], n_sample)
-        sample_size <- n_sample
+      if (n_distinct(df[[col]], na.rm=TRUE) <= 1) {
```

Comment:
Is there a case n_distinct(x) <1  ?

Reply:
It can happen when all values are NAs.

Reply:
Added comment.

Reply:
Would be great if you can add some comment on that.